src/ledger/insert_shred.zig

```diff
@@ -677,6 +733,8 @@ pub const ShredInserter = struct {
                     "Received *last* shred index {} less than previous shred index {}, and slot {} is not full, marking slot dead",
                     .{ shred_index, slot_meta.received, slot },
                 );
+                // QUESTION: Should the function not return at this point?
+                // Given we have a dead slots and we have saved that.
```

Comment:
That's a good point. It would probably help performance to return immediately. 

My main hesitation is a fear of diverging from agave's behavior, since agave continues to insert the shred at this point. I'm not sure whether or not it matters. I'd prefer to diverge only if I'm sure if it matters. It's probably a good idea to record ideas like this in a permanent document.

Maybe it could be beneficial to go ahead and insert the shred anyway. For example, maybe we do get all the shreds for the current slot, and consensus decides to actually use the larger version of the block. Then we would want to make sure we're not discarding shreds from that block. I'm not sure how likely this scenario is.